src/audio/google_rtc_audio_processing_mock.c

```diff
@@ -53,6 +91,35 @@ void GoogleRtcAudioProcessingFree(GoogleRtcAudioProcessingState *state)
 	}
 }
 
+int GoogleRtcAudioProcessingSetStreamFormats(GoogleRtcAudioProcessingState *const state,
+					     int capture_sample_rate_hz,
+					     int num_capture_input_channels,
+					     int num_capture_output_channels,
+					     int render_sample_rate_hz,
+					     int num_render_channels)
+{
+	rfree(state->aec_reference);
+
+	s->num_capture_channels = num_capture_input_channels;
+	s->num_aec_reference_channels = num_render_channels;
+	s->num_output_channels = num_capture_output_channels;
+	s->num_frames = capture_sample_rate_hz * 10 / 1000;
+	s->aec_reference = rballoc(0,
+				   SOF_MEM_CAPS_RAM,
+				   sizeof(s->aec_reference[0]) *
+				   s->num_frames *
+				   s->num_aec_reference_channels
+				   );
+	return 0;
```

Comment:
this looks very similar to above, can we maybe reuse it with helpers or inlines to mask out variables?